src/AnnotatedCode.c

```diff
@@ -73,3 +101,63 @@ R_API void r_annotated_code_print_json(RAnnotatedCode *code)
 	r_cons_printf ("%s\n", pj_string (pj));
 	pj_free (pj);
 }
+
+R_API void r_annotated_code_print_with_syntax_highlighting(RAnnotatedCode* code, RCore *core)
+{
+    if (code->annotations.len == 0) {
+        r_cons_printf("%s\n", code->code);
+        return;
+    }
+
+
+    size_t cur = 0;
+    size_t len = strlen(code->code);
+
+	RCodeAnnotation *annotation;
+	r_vector_foreach (&code->annotations, annotation) {
+        if (annotation->type != R_CODE_ANNOTATION_TYPE_SYNTAX_HIGHLIGHT)
+            continue;
+
+        // (1/3)
+        // now we have a syntax highlighting annotation.
+        // pick a suitable color for it.
+#define PALETE(x) (core->cons && core->cons->context->pal.x)? core->cons->context->pal.x 
+        const char* color = Color_RESET;
+        switch (annotation->syntax_highlight.type) {
+        case R_SYNTAX_HIGHLIGHT_TYPE_COMMENT:
+            color = PALETE(comment): Color_WHITE;
+            break;
+        case R_SYNTAX_HIGHLIGHT_TYPE_KEYWORD:
+            color = PALETE(pop): Color_MAGENTA;
```

Comment:
Hmm yeah I can't find any fitting entry too. I'd prefer to avoid adding a new entry if possible because all themes have to be adapted too, but if necessary it can be done. @radare what do you think?